decoders/svg/src/main/java/com/squareup/picasso3/decoder/svg/SvgImageDecoder.java

```diff
@@ -0,0 +1,52 @@
+package com.squareup.picasso3.decoder.svg;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import com.caverock.androidsvg.SVG;
+import com.caverock.androidsvg.SVGParseException;
+import com.squareup.picasso3.ImageDecoder;
+import com.squareup.picasso3.Request;
+import java.io.IOException;
+import okio.BufferedSource;
+
+class SvgImageDecoder implements ImageDecoder {
+
+  @Override public boolean canHandleSource(BufferedSource source) {
+    try {
+      SVG.getFromInputStream(source.peek().inputStream());
+      return true;
+    } catch (SVGParseException e) {
+      return false;
+    }
+  }
+
+  @Override public Image decodeImage(BufferedSource source, Request request) throws IOException {
```

Comment:
Would that simplify things? 

In the SVG case it would remove a parse, since we can't currently detect an SVG without parsing it, but I think that's sort of a special case. Bitmap, where we just decode the bounds to identify if it's parseable, doesn't have that limitation, and also nothing useful to return.  In those cases I think this dead simple API is nice, though I'll admit I'm not entirely sure what case #2 would look like.

Reply:
The bounds are useful to propagate for Bitmap so we can avoid re-reading them to apply transformations during decode.

Case #2 would work like Retrofit Converter.Factory or CallAdapter.Factory or Moshi JsonAdapter.Factory. If you can handle, return a handler.